Fix dynamic shapes repordering bug

tugsbayasgalan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #149528


WHen we create constraints, we look at the ordering of kwargs according to model signature. But when we trace, we use the ordering that is created based on how user passes in their kwargs. As a result, constraints and dynamic shapes end up having a different order causing issues when they have different dynamic tensor specs. 

Differential Revision: [D71478578](https://our.internmc.facebook.com/intern/diff/D71478578)

tugsbayasgalan committed:
Fix dynamic shapes repordering bug

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D71478578).

ydwu4 commented on torch/export/_trace.py:
```diff
@@ -1207,10 +1212,26 @@ def _get_range_constraints(
+        mod, args, kwargs, _is_torch_jit_trace=_is_torch_jit_trace
+    )
+
+    # This is because we trace based on the kewargs passed in from user
```
nit: kewargs -> kwargs

ydwu4 commented on torch/export/_trace.py:
```diff
@@ -1207,10 +1212,26 @@ def _get_range_constraints(
+    # This is because we trace based on the kewargs passed in from user
+    # not based on the signature. I feel it would be better to just enforce
+    # one ordering at the start of tracing to avoid confusions, but that is 
+    # bigger refactor, so do this to unblock for now.
```
lol, maybe left as TODO for BE week?

tugsbayasgalan committed:
Update on "Fix dynamic shapes repordering bug"



WHen we create constraints, we look at the ordering of kwargs according to model signature. But when we trace, we use the ordering that is created based on how user passes in their kwargs. As a result, constraints and dynamic shapes end up having a different order causing issues when they have different dynamic tensor specs. 

Differential Revision: [D71478578](https://our.internmc.facebook.com/intern/diff/D71478578)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D71478578).

tugsbayasgalan committed:
Update on "Fix dynamic shapes repordering bug"



WHen we create constraints, we look at the ordering of kwargs according to model signature. But when we trace, we use the ordering that is created based on how user passes in their kwargs. As a result, constraints and dynamic shapes end up having a different order causing issues when they have different dynamic tensor specs. 

Differential Revision: [D71478578](https://our.internmc.facebook.com/intern/diff/D71478578)

[ghstack-poisoned]
